muffnn/base.py

```diff
@@ -86,80 +86,133 @@ def fit(self, X, y):
         """
         _LOGGER.info("Fitting %s", re.sub(r"\s+", r" ", repr(self)))
 
-        # Check that the input is an array or sparse matrix.
-        # Convert to CSR if it's in another sparse format.
-        X, y = check_X_y(X, y, accept_sparse='csr', multi_output=True)
-        if y.ndim == 2 and y.shape[1] == 1:
-            # Following
-            # https://github.com/scikit-learn/scikit-learn/blob/51a765a/sklearn/ensemble/forest.py#L223,
-            # issue a warning if an Nx1 array was provided.
-            warn("A column-vector y was passed when a 1d array was"
-                 " expected. Please change the shape of y to "
-                 "(n_samples,), for example using ravel().",
-                 DataConversionWarning, stacklevel=2)
-            y = y[:, 0]
+        # Mark the model as not fitted (i.e., not fully initialized based on
+        # the data).
+        self._fitted = False
+
+        # Set the target preprocessing variables, which could otherwise be set
+        # based on the classes argument to partial_fit.
+        self._fit_targets(y)
```

Comment:
ah, thanks. I initially thought this line was needed here because fitting the target-related variables was tricky with the base class and subclasses. i agree that it's unnecessary and am removing it